test/extended/csrapprover/csrapprover.go

```
@@ -0,0 +1,136 @@
+		}
+
+		csrbytes, err := x509.CreateCertificateRequest(rand.Reader, certRequestTemplate, priv)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// get the token of the node-bootstrapper and use it to build a client for it
+		bootstrapperToken, err := oc.AsAdmin().Run("sa").Args("get-token", "node-bootstrapper", "-n", "openshift-machine-config-operator").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		saClientConfig := restclient.AnonymousClientConfig(oc.AdminConfig())
+		saClientConfig.BearerToken = bootstrapperToken
+
+		bootstrapperClient, err := kubeclient.NewForConfig(saClientConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		bootstrapperClient.CertificatesV1beta1().CertificateSigningRequests().Create(&certv1beta1.CertificateSigningRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "some-csr",
+			},
+			Spec: certv1beta1.CertificateSigningRequestSpec{
+				Request: pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE REQUEST", Bytes: csrbytes}),
+				Usages: []certv1beta1.KeyUsage{
+					certv1beta1.UsageDigitalSignature,
+					certv1beta1.UsageKeyEncipherment,
+					certv1beta1.UsageClientAuth,
+				},
+			},
+		})
+
+		csrClient := oc.AdminKubeClient().CertificatesV1beta1().CertificateSigningRequests()
+		defer cleanupCSR(csrClient, "some-csr")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = waitCSRStatus(csrClient)
+		// if status did not change in 30 sec, the CSR is still in pending
+		// which is fine as the machine-approver does not deny
+		if err.Error() == "CSR for unknown node should not be approved" {
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+	})
+})
+
+// waits for the CSR object to change status, checks that it did not get approved
+func waitCSRStatus(csrAdminClient certclientv1beta1.CertificateSigningRequestInterface) error {
+	return wait.Poll(1*time.Second, 30*time.Second, func() (bool, error) {
+		csr, err := csrAdminClient.Get("some-csr", metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) && err != nil {
```

Actually do not ignore any errors here.